kernel_gateway/services/kernels/handlers.py

```diff
@@ -51,6 +54,58 @@ def get_json_body(self):
                 model.setdefault('name', self.settings['kg_default_kernel_name'])
         return model
 
+class KernelHandler(TokenAuthorizationMixin,
+                    CORSMixin,
+                    notebook_handlers.KernelHandler):
+    '''This class will remove a kernel from the activity store when it is delted.
+    '''
+    @json_errors
+    def delete(self, kernel_id):
+        if self.settings.get('kg_list_kernels', False) == True:
+            activity.remove(kernel_id)
+        super(KernelHandler, self).delete(kernel_id)
+
+class ZMQChannelsHandler(TokenAuthorizationMixin,
+                        notebook_handlers.ZMQChannelsHandler):
+    '''This class listens for messages coming to and from the kernel to provide metrics
+    about kernel usage.
+    '''
+    def open(self, kernel_id):
+        if self.settings.get('kg_list_kernels', False) == True:
```

Comment:
`if self.settings.get('kg_list_kernels')` would be enough here. And in the other uses of this as well.